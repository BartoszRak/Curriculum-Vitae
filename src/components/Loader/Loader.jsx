import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'
import { LinearProgress } from '@material-ui/core'

import useStyleWithProps from './Loader.style'

export function Loader({ className, disabled, fill, children, size }) {
  const classes = useStyleWithProps({ size })
  const barProps = {
    'aria-hidden': true,
    classes: {
      root: classes.linearBar,
    },
    role: 'progressbar',
  }

  return disabled ? (
    children
  ) : (
    <div
      aria-busy
      aria-label="Loading content"
      className={classNames(classes.root, { [classes.fill]: fill }, className)}
    >
      <LinearProgress {...barProps} />
      <LinearProgress {...barProps} variant="query" />
      <LinearProgress {...barProps} />
      <LinearProgress {...barProps} variant="query" />
    </div>
  )
}

Loader.propTypes = {
  children: PropTypes.oneOfType([PropTypes.node, PropTypes.func]),
  className: PropTypes.string,
  disabled: PropTypes.bool,
  fill: PropTypes.bool,
  size: PropTypes.number,
}

Loader.defaultProps = {
  children: null,
  size: 2,
}

export default Loader
