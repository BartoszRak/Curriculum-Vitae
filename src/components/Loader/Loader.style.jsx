import { makeStylesWithProps } from '~hooks'

export default makeStylesWithProps((theme, { size }) => ({
  fill: {
    flex: '1 1 auto',
    height: '100%',
  },
  linearBar: {
    height: size,
    margin: '2px 0',
  },
  root: {
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
  },
}))
