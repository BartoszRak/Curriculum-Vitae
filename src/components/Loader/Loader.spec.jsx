import React from 'react'
import { shallow } from 'enzyme'

import { Loader } from './Loader'

const initComponent = overrides => {
  const mockProps = {
    children: <div id="testChildren">Test children</div>,
    classes: {},
  }
  const mockMethods = {}
  const wrapper = shallow(<Loader {...mockProps} {...mockMethods} {...overrides} />)
  return { mockProps, wrapper }
}

describe('global: Loader', () => {
  describe('rendering', () => {
    it('renders without crashing', () => {
      const { wrapper } = initComponent()
      expect(wrapper).toBeTruthy()
    })

    it('should render as expected', () => {
      const { wrapper } = initComponent()
      expect(wrapper).toMatchSnapshot()
    })

    it('should render as expected when not loading', () => {
      const { wrapper } = initComponent({ disable: true })
      expect(wrapper).toMatchSnapshot()
    })

    it('should render children when loading disabled', () => {
      const { wrapper } = initComponent({ disable: true })
      expect(wrapper.find('#testChildren').length).toBe(1)
    })

    it('should not render children when loading disabled', () => {
      const { wrapper } = initComponent({ disable: false })
      expect(wrapper.find('#testChildren').length).toBe(0)
    })
  })
})
