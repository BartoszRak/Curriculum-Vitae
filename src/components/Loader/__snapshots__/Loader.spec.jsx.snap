// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`global: Loader rendering should render as expected 1`] = `
<div
  aria-busy={true}
  className=""
  role="progressbar"
>
  <LinearProgressMock
    classes={
      Object {
        "root": undefined,
      }
    }
  />
  <LinearProgressMock
    classes={
      Object {
        "root": undefined,
      }
    }
    variant="query"
  />
  <LinearProgressMock
    classes={
      Object {
        "root": undefined,
      }
    }
  />
  <LinearProgressMock
    classes={
      Object {
        "root": undefined,
      }
    }
    variant="query"
  />
</div>
`;

exports[`global: Loader rendering should render as expected when not loading 1`] = `
<div
  id="testChildren"
>
  Test children
</div>
`;
